src/boosting/gbdt_model_text.cpp

```diff
@@ -621,6 +635,35 @@ bool GBDT::LoadModelFromString(const char* buffer, size_t len) {
   parser_config_str_ = ss.str();
   ss.clear();
   ss.str("");
+
+  int index = -1;
+  Str2Num tmp =  Str2Num();
+  while (p < end) {
+    auto line_len = Common::GetLine(p);
+    if (line_len > 0) {
+      std::string cur_line(p, line_len);
+      if (cur_line == std::string("Maps:")) {
+        is_inmaps = true;
+      } else if (cur_line == std::string("end of Maps")) {
+        p += line_len;
+        p = Common::SkipNewLine(p);
+        break;
+      } else if(cur_line == std::string("map")){
+        index++;
+        tmp.Str2NumMap.clear();
+      }
+       else if (is_inmaps) {
+         std::vector< std::string > key_value;
+         key_value = Common::Split(cur_line.c_str(), "=");
```

Comment:
I guess here you need some checks. One is that you may need to check if the key_value vector has two elements so your accessing key_value[1] won't bring exceptions. Another one is to do check if std::stod throws exception or not.